module3/lessons/Week2/IntroToMVC.md

```diff
@@ -0,0 +1,307 @@
+
+`InvalidOperationException: The view 'Index' was not found. The following locations were searched: /Views/Movies/Index.cshtml /Views/Shared/Index.cshtml`
+
+We know from the Request and Response cycle lesson that our application needs to return something in response to the HTTPS request. We are trying to return `return View();`, but we haven't yet created that view!
+
+## 3. Add a View
+
+View templates are created using Razor.
+
+Razor-based view templates:
+* Have a .cshtml file extension.
+* Provide an elegant way to create HTML output with C#.
+
+We will use Visual Studio to create a new view. Let's all follow these steps, but instead of calling our folder `HelloWorld` we will call it `Movies`.
+
+![Adding a View](/assets/images/module3/Week2/MVC_5_Create_View.png)
+
+Replace the contents of the `Views/Movies/Index.cshtml` Razor view file with the following:
+
+```C#
+<h1>Movies</h1>
+
+<p>All of the movies data will be shown here!</p>
+```
+<!-- TODO, add some content here about hot reloading. I think you we want to use the fire icon to enable reload on save? -->
+
+Navigate again to `https://localhost:<port#>/Movies`. The error should be gone, and you should now see "Movies All of the movies data will be shown here!"
+
+The `Index` method in the `MoviesController` ran the statement `return View();`, which specified that the method should use a view template file to render a response to the browser.
+
+A view template file name wasn't specified, so MVC defaulted to using the default view file. The default view has the same name as the action method, `Index` in this example. The view template `/Views/Movies/Index.cshtml` is used.
+
+## 4. Controller and View Practice
+<!-- TODO: Add practice here where students make another named route in their controller and another view with that same name. -->
+
+> With your partner: Create create a new route in your controller so that when the user visits `https://localhost:<port#>/cart` they see the string "Here are all of the items in your shopping cart"
+
+## 5. Add a Model
+
+These model classes are used with Entity Framework Core (EF Core) to work with a database.
+
+Let's start by adding a new model class.
+![Adding a model](/assets/images/module3/Week2/MVC_7_Adding_Model.png)
+
+Update the Models/Movie.cs file with the following code:
+
+```C#
+namespace MvcMovie.Models
+{
+    public class Movie
+    {
+        public int Id { get; set; }
+        public string Title { get; set; }
+        public string Genre { get; set; }
+    }
+}
+```
+The Movie class contains an Id field, which is required by the database for the primary key.
+
+### Setting up our DB
+
+Our next step is to add our `Movie` model to our database context.
+
+Update `DataAccess/MvcMovieContext` to the following:
+```C#
+using Microsoft.EntityFrameworkCore;
+using MvcMovie.Models;
+
+namespace MvcMovie.DataAccess
+{
+    public class MvcMovieContext : DbContext
+    {
+        public DbSet<Movie> Movies { get; set; }

```

Comment:
This is the only new line. Is there a way to highlight changes / new lines to the code so we can show what's being changed or added?

Reply:
I was thinking about that too, I tried doing a //This line is new later in the lesson. I agree some pattern for what should change would be helpful.